Sources/BSON/Document.swift

```diff
@@ -1,77 +1,184 @@
 import Foundation
 import NIO
 
+/// This shared allocator instance should be used for all underlying `ByteBuffer` creation.
+private let BSON_ALLOCATOR = ByteBufferAllocator()
+/// Maximum BSON document size in bytes
+private let BSON_MAX_SIZE = 0x100000
+
+extension ByteBuffer {
+    @discardableResult
+    internal mutating func writeCString(_ string: String) -> Int {
+        let written = self.writeString(string + "\0")
+        return written
+    }
+
+    internal mutating func readCString() throws -> String {
+        let string = try self.getCString(at: self.readerIndex)
+        self.moveReaderIndex(forwardBy: string.utf8.count + 1)
```

Comment:
do this and `getCString` need to be two separate methods? I only see `getCString` being used here